File: touralbum/app/src/main/java/com/example/touralbum/ui/reminder/base/BaseActivity.java
package com.example.touralbum.ui.reminder.base;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import butterknife.ButterKnife;
import butterknife.Unbinder;

public abstract class BaseActivity extends AppCompatActivity {
    private Unbinder mUnbinder;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getContentView());
        mUnbinder = ButterKnife.bind(this);
        initView();
        initData();
        setListener();
    }

    protected abstract void setListener();

    protected abstract void initView();

    protected abstract void initData();

    public abstract int getContentView();

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mUnbinder.unbind();
    }
}

File: touralbum/app/src/main/java/com/example/touralbum/ui/reminder/base/BaseActivity.kt
package com.example.touralbum.ui.reminder.base

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import butterknife.ButterKnife
import butterknife.Unbinder

abstract class BaseActivity : AppCompatActivity() {
    private var mUnbinder: Unbinder? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(contentView)
        mUnbinder = ButterKnife.bind(this)
        initView()
        initData()
        setListener()
    }

    protected abstract fun setListener()
    protected abstract fun initView()
    protected abstract fun initData()
    abstract val contentView: Int
    override fun onDestroy() {
        super.onDestroy()
        mUnbinder!!.unbind()
    }
}